config-linux.md

```diff
@@ -189,6 +189,91 @@ In addition to any devices configured with this setting, the runtime MUST also s
 * [`/dev/ptmx`][pts.4].
   A [bind-mount or symlink of the container's `/dev/pts/ptmx`][devpts].
 
+## <a name="configLinuxNetworkDevices" />Network Devices
+
+Linux network devices are entities that send and receive data packets.
+They are not represented as files in the /dev directory, unlike block devices, network devices are represented with the [`net_device`][net_device] data structure in the Linux kernel.
+Network devices have their own network namespace and a set of operations distinct from regular file operations. Examples of network devices include Ethernet cards, loopback devices, and virtual devices like bridges, VLANs, and MACVLANs.
+
+This schema focuses solely on moving existing network devices identified by name into the container namespace. It does not cover the complexities of network device creation or network configuration, such as IP address assignment, routing, and DNS setup.
+
+**`netDevices`** (object, OPTIONAL) set of network devices that MUST be available in the container. The runtime is responsible for providing these devices; the underlying mechanism is implementation-defined.
+
+Only privileged containers with a dedicated network namespace can have network devices directly assigned to them. This is required because moving network devices requires CAP_NET_ADMIN capabilities, not present on rootless containers, and to ensure security and avoid conflicts manipulate interfaces in the runtime network namespace.
```

Comment:
@aojea Could you explain why would a container need this capability to receive network devices?

The runtime need this capability to move the interface into the receiving netns, but that's about it. Of course if the container wants to (re-)configure an interface, it needs this capability, but that's out of scope. So, I'm not sure to get why a receiving container would need to be privileged.

Reply:
that statement is more about rootless containers implementation, where runtime itself doesn't have enough privileges to deal with network interfaces.

Reply:
Got it, thanks.

In that case what privileges a rootless runtime has is out-of-scope IMHO. Looking at this repo, it seems there's no single mention of what a 'rootless runtime' is. That would be a first one, and I don't think that's the right place to introduce this concept. For instance, is it what rootlesskit does? Is it a runtime with some specific capabilities dropped? etc...

I think the `netDevices` feature flag is just enough. If a runtime, rootless or not, doesn't support that feature, it should indicate that. Be it because of a lack of permission, or due to other reasons.

Reply:
yeah, this was added beacause of this comment about rootless containers https://github.com/opencontainers/runtime-spec/pull/1271#discussion_r1838850463

Should I just remove this paragraph or reword it @akerouanton ?

Reply:
> I think the netDevices feature flag is just enough. If a runtime, rootless or not, doesn't support that feature, it should indicate that.

removed the text that introduces the rootless concept